docs/lang/articles/basic/field.md

```diff
@@ -3,103 +3,128 @@ sidebar_position: 4
 ---
 
 # Fields
-Taichi fields are used to store data.
-In general, fields are global data containers that can be read and written from both the Python scope and the Taichi scope.
 
-A field has its own data type and shape and can be considered as a multi-dimensional array of elements.
-An element of a field can be a **scalar**, a **vector**, a **matrix**, or a **struct**.
-The sparsity of a field element is **dense** by default, but it can also be **sparse**, as detailed described in [Sparse spatial data structures](/lang/articles/advanced/sparse.md).
+The term _field_ is borrowed from mathematics and physics. If you already know [scalar field](https://en.wikipedia.org/wiki/Scalar_field) (for example heat field), or vector field (for example [gravitational field](https://en.wikipedia.org/wiki/Gravitational_field)), then it is straightforward for you to understand fields in Taichi.
 
-:::note
-The term **field** is borrowed from mathematics and physics.
-If you have already known [scalar field](https://en.wikipedia.org/wiki/Scalar_field) (e.g., heat field) or vector field (e.g., [gravitational field](https://en.wikipedia.org/wiki/Gravitational_field)) in mathematics and physics,
-it will be straightforward to understand the fields in Taichi.
-:::
+Fields in Taichi are the global data containers that can be accessed from both the Python scope and the Taichi scope. Just like an ndarray in NumPy or a tensor in PyTorch, a field in Taichi is defined as a multi-dimensional array of elements. Each of its elements can be a scalar, vector, matrix, struct, or pointer.
+
+- You can use a field as a sparse data structure, with each of its elememt being a pointer pointing to a different field either sparse or dense. And this pattern can go on. See the [Sparse spatial data structures](/lang/articles/advanced/sparse.md) for more information.
+- You can arrange fields using `ti.root` and `ti.axes`, for example stack one field on another or juxtapose them. All these are done at the memory level.
 
 ## Scalar fields
-We start introducing fields from this very basic type, the elements of scalar fields are simply scalars.
-* A 0D scalar field is a single scalar.
-* A 1D scalar field is an array.
-* A 2D scalar field can be used to represent a 2D regular grid of values.
-* A 3D scalar field can be used for volumetric data.
+
+Scalar fields refer to the fields that store scalars and are the most basic fields.
+
+- A 0D scalar field is a single scalar.
```

Comment:
+1. I think we should explain 0-D field here, because of its unique access pattern `f[None]`. For others, we can just name them N-D fields, meaning that their `shape` has N dimension

Reply:
Not fixed?

Reply:
- Line 12: Added a definition of 0D field: A 0D (one-dimensional) field contains *only* one element.
- Line 106 - 111: 
he following code snippet declares a 3D field of 2D vectors: 

```python
# Declare a 1x2x3 vector field, whose vector dimension is n=2
f = ti.Vector(ti.f32, n=2).field(shape=(1,2,3))
```

Reply:
I mean you don't need a bullet list to mention every dim from 1 to 3? 0D is a special case, others are the norm